apps/examples/fs_helper/fs_rw_test.c

```diff
@@ -0,0 +1,221 @@
+ * fs_rw_test
+ ****************************************************************************/
+
+static inline long get_time(void)
+{
+	struct timeval tv;
+
+	gettimeofday(&tv, NULL);
+	return (tv.tv_sec * 1000 + tv.tv_usec / 1000);
+}
+
+static void rw_testlog(char *printopt, const char *format, ...)
+{
+	va_list ap;
+
+	if (printopt[0] == 'y' || printopt[0] == 'Y') {
+		va_start(ap, format);
+		vfprintf(stdout, format, ap);
+		va_end(ap);
+	}
+}
+
+static int t_open(const char *path, int oflags, char *printopt)
+{
+	int fd = -1;
+	long t = 0;
+	long dt = 0;
+	char mode = 0;
+
+	if ((oflags & O_RDONLY) != 0) {
+		mode = 'r';
+	} else if ((oflags & O_WRONLY) != 0) {
+		mode = 'w';
+	} else {
+		printf("open fail please select O_RDONLY or O_WRONLY option\n");
+		return -1;
+	}
+
+	t = get_time();
+	fd = open(path, oflags);
+	if (fd == -1) {
+		printf("%s errno: %d\n", __FUNCTION__, errno);
+		exit(EXIT_FAILURE);
+	}
+	dt = get_time() - t;
+	total_time += dt;
+	rw_testlog(printopt, "open(%c):%dms, ", mode, dt);
+	return fd;
+}
+
+static void t_write(int fd, const void *buf, size_t nbytes, char *printopt)
+{
+	long t = 0;
+	long dt = 0;
+	int ret = 0;
+
+	t = get_time();
+	ret = write(fd, buf, nbytes);
+	if (ret == -1) {
+		printf("%s errno: %d\n", __FUNCTION__, errno);
+		exit(EXIT_FAILURE);
+	}
+	dt = get_time() - t;
+	total_time += dt;
+	rw_testlog(printopt, "write:%dms, ", dt);
+}
+
+static void t_close(int fd, char *printopt)
+{
+	long t = 0;
+	long dt = 0;
+	int ret = 0;
+
+	t = get_time();
+	ret = close(fd);
+	if (ret == -1) {
+		printf("%s errno: %d\n", __FUNCTION__, errno);
+		exit(EXIT_FAILURE);
+	}
+	dt = get_time() - t;
+	total_time += dt;
+	rw_testlog(printopt, "close:%dms\n", dt);
+}
+
+static void t_read(int fd, char *buf, size_t nbytes, char *printopt)
+{
+	long t = 0;
+	long dt = 0;
+	int ret = 0;
+
+	t = get_time();		
+	ret = read(fd, buf, nbytes);
+	if (ret == -1) {
+		printf("%s errno: %d\n", __FUNCTION__, errno);
+		exit(EXIT_FAILURE);
+	}
+	dt = get_time() - t;
+	total_time += dt;
+	itoa((int)buf[0], buf, 10);
+	rw_testlog(printopt, "read:%dms, data:%s ", dt, buf);
+}
+
+static void rw_operation(char *path, int size, int operation, int iteration, char *printopt)
+{
+	int fd = -1;
+	char data[size];
+	char buf[size];
+
+	data[0] = iteration;
+	rw_testlog(printopt, "==========iteration %d==========\n", iteration);
+	switch (operation) {
+	case 1:
+		fd = t_open(path, O_WRONLY | O_CREAT | O_TRUNC, printopt);
+		t_write(fd, data, sizeof(data), printopt);
+		t_close(fd, printopt);
+		fd = t_open(path, O_RDONLY, printopt);
+		t_read(fd, buf, sizeof(buf), printopt);
+		break;
+	case 2:
+		fd = t_open(path, O_RDONLY, printopt);
+		t_read(fd, buf, sizeof(buf), printopt);
+		break;
+	case 3:
+		fd = t_open(path, O_WRONLY | O_CREAT | O_TRUNC, printopt);
+		t_write(fd, data, sizeof(data), printopt);
+		break;
+	default:
+		return;
+	}
+	t_close(fd, printopt);
+	
```

Comment:
there is a space at 180 line please remove it

Reply:
modified